### DISCUSSION TITLE
N-SPC R&D1 and Konami Variant Updates

### DISCUSSION THREAD

**Discussion by KungFuFurby** [vgmtrans/vgmtrans#227] [Music formats]
Here are some differences and identifications for existing N-SPC/Kankichi variants that I caught that aren't implemented in VGMTrans yet:

I have identified the R&D1 variant unknown commands as following (these carry over into Lemmings)...
* VCMD $FB is a NOP, skipping two parameter bytes.
* VCMD $FC disables playing back subsequent notes for a channel.
* VCMD $FD activates fast forward. This causes the player to rapidly progress throughout the song until it is turned off.
* VCMD $FE deactivates fast forward.

In the Konami variant...
* The tempo fade command (VCMD $E8) is restored from the standard variant exclusively for Akumajou Dracula/Super Castlevania IV. It is a NOP in all other cases via an immediate RET.
* VCMD $E4 is identified as a strange pitch envelope in the disassemblies. I believe it is a precursor to the random pitch envelope used in later games.
* VCMDs $F5-$F8 disable echo writes, also coming from the disassemblies.

**Comment by loveemu** [comment]
Nice find! Thank you for reporting.
